packages/components/src/date-range-filter-picker/index.js

```diff
@@ -160,21 +157,26 @@ DateRangeFilterPicker.propTypes = {
 	 */
 	onRangeSelect: PropTypes.func.isRequired,
 	/**
-	 * The query string represented in object form.
+	 * The date query string represented in object form.
 	 */
-	query: PropTypes.object,
-	/**
-	 * Store date utility instance.
-	 */
-	storeDate: PropTypes.shape( {
-		isoDateFormat: PropTypes.string.isRequired,
-		getDateParamsFromQuery: PropTypes.func.isRequired,
-		getCurrentDates: PropTypes.func.isRequired,
+	dateQuery: PropTypes.shape( {
+		period: PropTypes.string.isRequired,
+		compare: PropTypes.string.isRequired,
+		before: PropTypes.instanceOf( moment ),
+		after: PropTypes.instanceOf( moment ),
+		primaryDate: PropTypes.shape( {
+			label: PropTypes.string.isRequired,
+			range: PropTypes.string.isRequired,
+			after: PropTypes.instanceOf( moment ).isRequired,
+			before: PropTypes.instanceOf( moment ).isRequired,
```

Comment:
To avoid importing moment, can you type check against `object` here instead and then add defensive programming elsewhere in whatever component is actually _implementing_  `before`/`after` (i.e. [you could check if `moment.isMoment` is present and `true`](https://momentjscom.readthedocs.io/en/latest/moment/05-query/10-is-a-moment/)).

Reply:
It looks like we both misread/understood this interface. `moment.isMoment()` as referenced in the docs is a utility function on the exported `moment` module, not on an instance of `moment()`. It looks like we can look for a `_isAMomentObject` property instead.